src/stories/components/UploadPhoto/UploadPhoto.js

```diff
@@ -21,47 +21,66 @@ import ImageService from '../../../firebase/CRUDServices/imageService';
  */
 export const UploadPhoto = () => {
   const [cameraIsOn, setWebcamOn] = useState(false);
+  const [file, setFile] = useState('');
   const [image, setImage] = useState('');
+  const [error, setError] = useState(false);
   const webcamRef = React.useRef(null);
 
   // toggle device camera
   const toggleCamera = () => {
     setWebcamOn(!cameraIsOn);
   };
 
-  // take a photo
-  const capturePhoto = React.useCallback(
-    () => {
-      setImage(webcamRef.current.getScreenshot());
-    },
-    [webcamRef],
-  );
+  // take a photo via webcam
+  const capturePhoto = React.useCallback(() => {
+    setImage(webcamRef.current.getScreenshot());
+    document.getElementById('fileName').innerHTML = 'No file selected.';
+  }, [webcamRef]);
+
+  // use file input value
+  const uploadFile = (target) => {
+    // set file variable
+    setFile(target.files.item(0));
+    // reset image
+    setImage('');
+  };
 
+  // upload photo to the server to generate stimuli
   const uploadPhoto = () => {
-    const service = ImageService.getInstance();
+    // REMOVE EVENTUALLY
     const userId = 'test';
     const experimentId = '001';
-    service.postRawImage(userId, experimentId, image).then((response) => {
-      // REMOVE EVENTUALLY
-      window.alert(response);
-    });
-  };
-
-  const selectImage = (target) => {
-    setImage(target.files.item(0));
 
-    if (document.getElementById('file-upload__input')) {
-      const name = document.getElementById('file-upload__input');
-      const selectedFile = name.files.item(0).name;
-      document.getElementById('file-upload__selected').innerHTML = selectedFile;
+    if (image) {
+      fetch(image).then((response) => response.blob())
+        .then((blob) => {
+          const file = new File([blob], 'photo', { type: 'image/jpeg' });
+          setFile(file);
+        });
```

Comment:
Convert the webcam snapshot to a file.

Reply:
In the state:
```
const [file, setFile] = useState('');
const [image, setImage] = useState('');
```

Reply:
oh okay, like i was wondering if it was stored in browser memory or local storage but I think state is stored in local storage. Not a big deal I was just curious